app/dash_app.py

```diff
@@ -126,44 +132,91 @@ def _read_coherence(coherence_csv):
     return coh
 
 
+def _read_baseline(baseline_csv):
+    baseline = pd.read_csv(
+        baseline_csv,
+        delimiter=' ',
+        header=None,
+        skipinitialspace=True)
+    baseline.columns = ['index',
+                        'first_date',
+                        'second_date',
+                        'bperp',
+                        'btemp',
+                        'bperp2',
+                        'x']
+    baseline = baseline.drop(['index',
+                              'btemp',
+                              'bperp2',
+                              'x'], axis=1)
+    baseline['first_date'] = pd.to_datetime(baseline['first_date'],
+                                            format="%Y%m%d")
+    baseline['second_date'] = pd.to_datetime(baseline['second_date'],
+                                             format="%Y%m%d")
+    return baseline
+
+
 def _valid_dates(coh):
     return coh.first_date.dropna().unique()
 
 
 def _coherence_csv(target_id):
-    site, beam = target_id.split('_', 1)
+    site, beam = target_id.rsplit('_', 1)
     return f'Data/{site}/{beam}/CoherenceMatrix.csv'
 
 
+def _baseline_csv(target_id):
+    site, beam = target_id.rsplit('_', 1)
+    return f'Data/{site}/{beam}/bperp_all'
+
+
 def pivot_and_clean(coh_long):
     """Convert long-form coherence to wide-form and clean it up."""
     coh_wide = coh_long.pivot(
         index='delta_days',
         columns='second_date',
         values='coherence')
-
     # include zero baseline even though it will never be valid
     coh_wide.loc[0, :] = np.NaN
     coh_wide.sort_index(inplace=True)
-
     # because hovertemplate 'f' format doesn't handle NaN properly
     coh_wide = coh_wide.round(2)
-
     # trim empty edges
     coh_wide = coh_wide.loc[
         (coh_wide.index >= 0) &
         (coh_wide.index <= coh_wide.max(axis='columns').last_valid_index()),
         (coh_wide.columns >= coh_wide.max(axis='index').first_valid_index()) &
         (coh_wide.columns <= coh_wide.max(axis='index').last_valid_index())]
-
     return coh_wide
 
 
+def pivot_and_clean_dates(coh_long, coh_wide):
```

Comment:
Why is coh_wide an input argument? This should have shown up in a lint check as an unused variable. I think maybe we should start using mypy typing, to make things like this more readable.